result/result.mbt

```diff
@@ -11,3 +11,248 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/// Maps the value of a Result if it is `Ok` into another, otherwise returns the `Err` value unchanged.
+///
+/// # Example
+///
+/// ```
+/// let x = Ok(6)
+/// let y = x.map(fn (v : Int) { v * 7 })
+/// ```
+pub fn map[T, E, U](self : Result[T, E], f : (T) -> U) -> Result[U, E] {
+  Ok(f(self?))
+}
+
+test "map" {
+  let x : Result[Int, Unit] = Ok(6)
+  let y = x.map(fn(v : Int) { v * 7 })
+  @assertion.assert_eq(y, Ok(42))?
```

Comment:
We can also test the `Result::Err` case to ensure that the behavior of the function being tested does not change accidentally in the future.